usr/sbin/cloudguest-lic-watcher

```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/python3
+
+# Copyright (c) 2025, SUSE LLC, All rights reserved.
+#
+# This library is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License as published by the Free Software Foundation; either
+# version 3.0 of the License, or (at your option) any later version.
+# This library is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# Lesser General Public License for more details.
+# You should have received a copy of the GNU Lesser General Public
+# License along with this library.
+
+import logging
+import os
+
+import instance_billing_flavor_check.utils as inst_flvr_utils
+import cloudregister.registerutils as utils
+
+CACHE_LICENSE_PATH = os.path.join(utils.get_state_dir(), 'cached_license')
+
+def update_license_cache(license_type):
+    """Update the cache to track the license type"""
+    with open(CACHE_LICENSE_PATH, 'w+') as cached_license:
```

Comment:
why "w+" open for writing and reading when you only write to the file ?

Reply:
Yep, fixed.